src/Renderer/TaskLineRenderer.ts

```diff
@@ -21,6 +21,37 @@ export type TextRenderer = (
     obsidianComponent: Component | null, // null is allowed here only for tests
 ) => Promise<void>;
 
+/**
+ * Create an HTML element.
+ *
+ * Unlike the equivalent Obsidian convenience function li.createEl(),
+ * this can be called from our automated tests.
+ *
+ * @param parentUlElement
+ * @param tagName
+ * @param options - See the `document.createElement()` call in
+ *      [Web component example](https://developer.mozilla.org/en-US/docs/web/api/document/createelement#web_component_example)
+ *      for an example of use.
+ *
+ * @example <caption>Example call:</caption>
+ * const li = createElement(this.parentUlElement, 'li');
+ */
+export function createElement<K extends keyof HTMLElementTagNameMap>(
+    parentUlElement: HTMLElement,
```

Comment:
> The parameter name shall probably be `parentElement`, the `Ul` is not necessary here.

Done.

> I also suggest to move the tagName to the first paramer so it would be more readable: createElement('li',...)

I tried this via automated refactorings in WebStorm, and each time it made no changes.
So I tried manually moving the tagName to the first parameter, and it gave a really confusing error that tagName wasn't in JSDocs, when it was.

Turns out it's not possible with generic parameters:
https://youtrack.jetbrains.com/issue/RSRP-288544/Be-able-to-re-order-generic-type-parameters-when-performing-a-Change-Signature-refactoring

I tried doing it manually several times, and was just fighting WebStorm - the cursor moved unexpectedly.

In the end I didn't feel strongly enough about it to make the change manually.

Reply:
> I also suggest to move the tagName to the first paramer so it would be more readable: createElement('li',...)

I did manage to do that in the end, and it does read a lot better. Thanks!